WS2012R2/lisa/setupscripts/AddVhdxHardDisk.ps1

```diff
@@ -358,16 +358,11 @@ if ($testParams -eq $null -or $testParams.Length -lt 3)
     return $False
 }
 
-# WS 2012, 2008 R2 do not support generation 2 VMs
-$OSInfo = get-wmiobject Win32_OperatingSystem -ComputerName $hvServer
-if ( ($OSInfo.Caption -match '.2008 R2.') -or 
-     ($OSInfo.Caption -match '.2012 [^R2].')) {
+$vmGeneration = Get-VM $vmName -ComputerName $hvServer| select -ExpandProperty Generation -ErrorAction SilentlyContinue
```

Comment:
with codes below:
`361 $vmGeneration = Get-VM $vmName -ComputerName $hvServer| select -ExpandProperty Generation -ErrorAction SilentlyContinue
362 if ($? -eq "False")
363 {
364    $vmGeneration = 1
365 }
`
if os don't support generation property, it will return false we can set vmgeneration=1, or else it return ture, we can get the VM generation correctly. 
so I think we needn't check the version of OS, only check $? that is OK